sys/ztimer/Kconfig

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menuconfig MODULE_ZTIMER
+    bool "ztimer - High level timer abstraction layer"
+    depends on TEST_KCONFIG
+    select MODULE_ZTIMER_CORE
+    select MODULE_ZTIMER_CONVERT_FRAC
+    select MODULE_ZTIMER_CONVERT_SHIFT
+    select MODULE_ZTIMER_EXTEND
+
+if MODULE_ZTIMER
+
+menu "Backends"
+
+config MODULE_ZTIMER_PERIPH_RTC
+    bool "RTC peripheral"
+    depends on HAS_PERIPH_RTC
+    select MODULE_PERIPH_RTC
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_PERIPH_RTT
+    bool "RTT peripheral"
+    depends on HAS_PERIPH_RTT
+    select MODULE_PERIPH_RTT
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_PERIPH_TIMER
+    bool "Timer peripheral"
+    depends on HAS_PERIPH_TIMER
+    select MODULE_PERIPH_TIMER
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_MOCK
+    bool "Mock (for testing)"
+    select HAVE_ZTIMER_BACKEND
+    help
+        This ztimer module implements a virtual clock that can be used for
+        unittests. It can be manually adjusted to different timestamps and
+        manually fired to simulate different scenarios and test the ztimer
+        implementation using this as a backing timer.
+
+config HAVE_ZTIMER_BACKEND
+    bool
+    help
+        Indicates that a ztimer backend is provided.
+
+endmenu # Backends
+
+menu "Clocks"
+
+config MODULE_ZTIMER_USEC
+    bool "Microseconds"
+    depends on MODULE_ZTIMER_PERIPH_TIMER
+
+config MODULE_ZTIMER_MSEC
+    bool "Milliseconds"
+    depends on HAVE_ZTIMER_BACKEND
```

Comment:
What about the mock backend? Do you know which clocks that can enable?

Reply:
I think none. The `ZTIMER_USEC`, `ZTIMER_MSEC` and `ZTIMER_SEC` clocks are initialized in `ztimer_init()` in `sys/ztimer/auto_init.c` - but there the mock backend is not used.

The mock backed will only be used in tests for `ztimer`, where ztimer can be manually initialized by the test application. So the mock clock should IMO be hidden in `make menuconfig`.

Reply:
> So the mock clock should IMO be hidden in make menuconfig.

The problem is that if it is hidden then it can't be enabled via a `.config` file in the test application. What the user can see through `menuconfig` is the same that is configurable via a `.config` file. An extra symbol to indicate that a test is being run to make it visible only then could be added, but it seems a bit overkill. Maybe I can just move it out of this menu.

Reply:
Changed, now `ztimer_msec` depends on either having `ztimer_periph_timer` or `ztimer_periph_rtt`.